src/main/resources/com/owlcyberdefense/syslog/xsd/syslog-rfc5424.dfdl.xsd

```diff
@@ -0,0 +1,215 @@
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS WITH THE
+SOFTWARE.
+-->
+
+<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema"
+           xmlns:dfdl="http://www.ogf.org/dfdl/dfdl-1.0/"
+           xmlns:sl="urn:syslog"
+           targetNamespace="urn:syslog">
+
+  <xs:include schemaLocation="common.dfdl.xsd"/>
+
+  <xs:annotation>
+    <xs:appinfo source="http://www.ogf.org/dfdl/">
+      <dfdl:format ref="sl:common"/>
+    </xs:appinfo>
+  </xs:annotation>
+
+  <xs:element name="SyslogRFC5424">
+    <xs:annotation>
+      <xs:documentation>
+        Defines a single syslog message of the RFC5424 format, including the
+        facility and severity. This should be used as the root element if the
+        goal is to only support RFC5424 messages.
+      </xs:documentation>
+    </xs:annotation>
+    <xs:complexType>
+      <xs:sequence>
+        <xs:group ref="sl:Prival" />
+        <xs:group ref="sl:RFC5424" />
+      </xs:sequence>
+    </xs:complexType>
+  </xs:element>
+
+  <xs:complexType name="RFC5424">
+    <xs:annotation>
+      <xs:documentation>
+        Defines a type containing the content of a RFC5424 syslog message,
+        excluding the facility and severity. This should be used when the
+        faciliy/severity are already parsed.
+      </xs:documentation>
+    </xs:annotation>
+    <xs:sequence>
+      <xs:group ref="sl:RFC5424" />
+    </xs:sequence>
+  </xs:complexType>
+
+  <xs:group name="RFC5424">
+    <xs:annotation>
+      <xs:documentation>
+        Defines all content of a RFC5424 syslog message that appears immediately
+        affter the severify/facility part of a syslog message.
+      </xs:documentation>
+    </xs:annotation>
+    <xs:sequence>
+      <xs:element name="Header">
+        <xs:complexType>
+          <xs:group ref="sl:rfc5424headerGroup"/>
+        </xs:complexType>
+      </xs:element>
+      <xs:group ref="sl:rfc5424BodyGroup"/>
+    </xs:sequence>
+  </xs:group>
+
+  <xs:group name="rfc5424headerGroup">
+    <xs:sequence>
+      <xs:sequence dfdl:separator="%SP;" dfdl:separatorPosition="postfix">
+        <xs:element name="Version" type="xs:int" dfdl:textNumberPattern="#0">
+          <xs:annotation>
+            <xs:appinfo source="http://www.ogf.org/dfdl/">
+              <dfdl:discriminator test="{ . eq 1 }" />
+            </xs:appinfo>
+          </xs:annotation>
+        </xs:element>
+        <xs:choice>
+          <xs:element name="Timestamp" type="xs:dateTime" dfdl:calendarPattern="yyyy-MM-dd'T'HH:mm:ssXXX" nillable="true" />
+          <xs:element name="TimestampNano" type="xs:dateTime" dfdl:calendarPattern="yyyy-MM-dd'T'HH:mm:ss.SSSXXX" nillable="true" />
+        </xs:choice>
+        <xs:element name="Hostname" nillable="true">
+          <xs:simpleType>
+            <xs:restriction base="xs:string">
+              <xs:minLength value="1" />
+              <xs:maxLength value="255" />
+              <xs:pattern value="[!-~]+" />
```

Comment:
Replace "+" with "{1,255}". I know it's barfy from an engineering perspective because you have the 255 in multiple places then, but it will silence a warning from an XSAT-style checker that just greps for pattern facets containing * or +.